pkg/objstore/s3/s3_test.go

```diff
@@ -163,6 +163,63 @@ http_config:
 	}
 }
 
+func TestParseConfig_CustomHTTPConfigWithTLS(t *testing.T) {
+	input := []byte(`bucket: abcd
+insecure: false
+http_config:
+  tls_config:
+    ca_file: /certs/ca.crt
+    cert_file: /certs/cert.crt
+    key_file: /certs/key.key
+    server_name: server
+    insecure_skip_verify: false
+  `)
+	cfg, err := parseConfig(input)
+	testutil.Ok(t, err)
+
+	if cfg.HTTPConfig.TLSConfig.CAFile != "/certs/ca.crt" {
+		t.Errorf("parsing of ca_file failed: got %v, expected %v",
```

Comment:
Let's use `testutil.Equals(t, "/certs/ca.crt", cfg.HTTPConfig.TLSConfig.CAFile)` and so on.